derive/src/lib.rs

```diff
@@ -17,26 +17,204 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
 extern crate alloc;
-extern crate proc_macro;
 
 mod impl_wrapper;
-mod metadata;
-mod type_def;
-mod type_id;
 
+use alloc::vec::Vec;
 use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::{
+	parse::{Error, Result},
+	parse_quote,
+	punctuated::Punctuated,
+	token::Comma,
+	Data, DataEnum, DataStruct, DeriveInput, Expr, ExprLit, Field, Fields, Lit, Variant,
+};
```

Comment:
Have you considered using the https://crates.io/crates/synstructure crate?
I used it in `ink_core` for the derives there and it was an extreme pleasure to use after I got how to use the crate. The `synstructure` crate really goes into the extremes of generality in derive macros which was a nice experience to me.

Reply:
I did look at it, but it didn't seem suitable because looks to be designed to generate derives for instance methods, where you need to perform the operation on values bound from `match self`. Correct me if I'm wrong